src/attack.c

```diff
@@ -655,8 +664,22 @@ attack_host (struct scan_globals *globals, struct in6_addr *ip, GSList *vhosts,
           return;
         }
 
+      if (check_kb_inconsistency (globals, main_kb) != 0)
+        {
+          // As long as we don't have a proper communication channel
+          // to our ancestors we just kill our parent and ourselves
+          // (but let our grandparents live).
+          // To prevent duplicate results we don't let ACT_END run.
+          me = getpid ();
+          kill (parent, SIGQUIT);
+          // just in case
+          kill (me, SIGQUIT);
+          return;
```

Comment:
A few things about this part
1. The signal SIGQUIT will be ignored by the main openvas-process, which should be the parent here. Maybe a SIGUSR1 can be send instead, this would signal that the scan should be stopped.
2. Is there a reason why using SIGQUIT instead of SIGTERM?
3. To send a signal to the own process the function `raise` can be used